src/_guides/language/effective-dart/usage.md

```diff
@@ -354,13 +354,20 @@ then it probably does make sense to have a separate boolean field.
 
 Checking that a nullable variable is not equal to `null` promotes the variable
 to a non-nullable type. That lets you access members on the variable and pass it
-to functions expecting a non-nullable type. Unfortunately, promotion is only
-sound for local variables and parameters, so fields and top-level variables
-aren't promoted.
+to functions expecting a non-nullable type. 
 
-One pattern to work around this is to assign the field's value to a local
-variable. Null checks on that variable do promote, so you can safely treat
-it as non-nullable.
+Type promotion is only sound, however, for local variables, parameters, and
+private final fields. Instances that are open to manipulation, whether
+explicitly, like top-level variables, or implicitly, like fields with a concrete
+getter of the same name in the same library, cannot be type promoted.
+
+Declaring members [private] and [final] as we generally recommend would bypass
+these limitations, but that's not always an option. One pattern to work around
+this is to assign the field's value to a local variable. Null checks on that
+variable will promote, so you can safely treat it as non-nullable.
```

Comment:
**suggestion:** Tighten some of the language. The sentences run a little long.

```suggestion
You can _soundly_ promote the types of local variables, parameters, and private final fields only.
You cannot promote instances that can be manipulated.
This potential manipulation can be explicit, like top-level variables, or implicit,
like fields with a concrete getter of the same name in the same library.

To bypass these limitations, try one of the following options:

- Declare members [private] and [final].
- Assign the field's value to a local variable. Null checks on that
variable promote, so you can safely treat it as non-nullable.
```

Reply:
I'm gonna skip direct-committing this because there's a lot of nuance here that gets lost in these suggestions.

For example, "You can soundly promote..." and "You cannot promote..."; type promotion is handled by the compiler, so it's odd to attribute it to the reader/user. I guess you can technically write your code in a way that enables type promotion, but "you" cannot promote instances. 

Also, the paragraph starting on line 364 intentionally separates the points you combined into a bullet list. While they both technically make type promotion "work", they are conceptually different. The first point, declaring members private and final, 
1. is not covered in the example that immediately follows, nor is it the point of this section at all, and 
2. is what we want users to be doing all the time as a best practice. 

What I intended this paragraph to say was "if there's some reason you _can't_ use the best practice we want you to be using all the time, here's a different way to make type promotion work", not "here's two ways to make type promotion work". I think I'll just completely remove the first sentence though, seeing now how it can be misconstrued. 

I definitely agree the long sentence starting on line 360 is not very concise though (I spent a long, _long_ time getting these concepts compressed into a few sentences, so I do understand the immediate reaction to the wordiness). I think I'll end up moving that level of detail on type promotion constraints to "Understanding null safety" and just get rid of it all together here.